pkg/core/tekton/tekton.go

```diff
@@ -0,0 +1,405 @@
+	timeout := 1 * time.Minute
+	if err := waitFor(t.eventListenerReady(ctx, el.Name), interval, timeout); err != nil {
+		return "", fmt.Errorf("event listener %q did not get ready in the expected time", el.Name)
+	}
+
+	el, _ = t.tektonClients.EventListenerClient.Get(ctx, workflowName, metav1.GetOptions{})
+	return el.Status.Address.URL.String(), nil
+}
+
+func waitFor(waitFunc wait.ConditionFunc, interval, timeout time.Duration) error {
+	return wait.PollImmediate(interval, timeout, waitFunc)
+}
+
+func (t *WorkflowBackend) eventListenerReady(ctx context.Context, name string) wait.ConditionFunc {
+	return func() (bool, error) {
+		el, err := t.tektonClients.EventListenerClient.Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			return false, nil
+		}
+		// No conditions have been set yet
+		if len(el.Status.Conditions) == 0 {
+			return false, nil
+		}
+		if el.Status.GetCondition(apis.ConditionType(appsv1.DeploymentAvailable)) == nil {
+			return false, nil
+		}
+		for _, cond := range el.Status.Conditions {
+			if cond.Status != corev1.ConditionTrue {
+				return false, nil
+			}
+		}
+		if el.Status.Address.URL == nil {
+			return false, nil
+		}
+		return true, nil
+	}
+}
+
+func generatePipeline(w workflow.Workflow, namespace string) *v1beta1.Pipeline {
+	resolver := newVariablesResolver()
+	pb := builder.NewPipelineBuilder(w.Name, namespace)
+	pb.PipelineMeta(builder.Label("fuseml/generated-from", w.Name))
+	pb.PipelineDescription(*w.Description)
+
+	// process the FuseML workflow inputs
+	for _, input := range w.Inputs {
+		// an input of the 'codeset' type means a git repository for tekton.
+		// adds a workspace, resource and the clone task to the pipeline, also
+		// also add 'codeset-name' and 'codeset-version' parameters to the tekton
+		// pipeline which represents codeset.name and codeset.version in a FuseML
+		// workflow.
+		if *input.Type == inputTypeCodeset {
+			pb.PipelineWorkspace(codesetWorkspaceName)
+			pb.PipelineResource("source-repo", "git")
+			pb.PipelineTask("clone", cloneTaskName, nil, map[string]string{codesetWorkspaceName: codesetWorkspaceName},
+				map[string]string{"source-repo": "source-repo"})
+			pb.PipelineParam(codesetNameParam, "Reference to the codeset (git project)")
+			resolver.addReference(fmt.Sprintf("inputs.%s.name", *input.Name), fmt.Sprintf("$(params.%s)", codesetNameParam))
+			pb.PipelineParam(codesetVersionParam, "Codeset version (git revision)", "main")
+			resolver.addReference(fmt.Sprintf("inputs.%s.version", *input.Name), fmt.Sprintf("$(params.%s)", codesetVersionParam))
+		} else {
+			pb.PipelineParam(*input.Name, *input.Description, *input.Default)
+			resolver.addReference(fmt.Sprintf("inputs.%s", *input.Name), fmt.Sprintf("$(params.%s)", *input.Name))
+		}
+	}
+
+	// process the FuseML workflow steps
+STEPS:
+	for _, step := range w.Steps {
+		for _, output := range step.Outputs {
+			// image type output parameters serve as input for the tekton builder task (kaniko)
```

Comment:
I think it should be mentioned here or somewhere near that when you are processing the builder step, you are ignoring the "image" value defined in the workflow...